aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -150,6 +187,167 @@ module aptos_framework::coin {
     /// Capability required to burn coins.
     struct BurnCapability<phantom CoinType> has copy, store {}
 
+    /// The mapping between coin and fungible asset.
+    struct CoinConversionMap has key {
+        coin_to_fungible_asset_map: Table<TypeInfo, address>,
+    }
+
+    #[resource_group_member(group = aptos_framework::object::ObjectGroup)]
+    /// The paired coin type info stored in fungible asset metadata object.
+    struct PairedCoinType has key {
+        type: TypeInfo,
+    }
+
+    inline fun borrow_conversion_map_mut(): &mut CoinConversionMap {
+        if (!exists<CoinConversionMap>(
+            @aptos_framework
+        ) && std::features::coin_to_fungible_asset_migration_feature_enabled()) {
+            move_to(&create_signer::create_signer(@aptos_framework), CoinConversionMap {
+                coin_to_fungible_asset_map: table::new<TypeInfo, address>(),
+            })
+        };
+        borrow_global_mut<CoinConversionMap>(@aptos_framework)
+    }
+
+    public entry fun pair_existing_fungible_asset<CoinType>(
+        coin_creator: &signer,
+        metadata: Object<Metadata>
+    ) acquires CoinConversionMap, CoinInfo {
+        // Verify the metadata is the same
+        assert!(
+            name<CoinType>() == fungible_asset::name(metadata),
+            error::invalid_argument(ECOIN_AND_FUNGIBLE_ASSET_NAME_MISMATCH)
+        );
+        assert!(
+            symbol<CoinType>() == fungible_asset::symbol(metadata),
+            error::invalid_argument(ECOIN_AND_FUNGIBLE_ASSET_SYMBOL_MISMATCH)
+        );
+        assert!(
+            decimals<CoinType>() == fungible_asset::decimals(metadata),
+            error::invalid_argument(ECOIN_AND_FUNGIBLE_ASSET_DECIMAL_MISMATCH)
+        );
+
+        // Verify both track supply or neither tracks
+        assert!(
+            option::is_some(&coin_supply<CoinType>()) == option::is_some(&fungible_asset::supply(metadata)),
+            error::invalid_argument(ESUPPLY_TRACKING_MISMATCH)
+        );
+
+        let map = borrow_conversion_map_mut();
+        let type = type_info::type_of<CoinType>();
+        let metadata_addr = object::object_address(&metadata);
+        assert!(!table::contains(&map.coin_to_fungible_asset_map, type), error::already_exists(EPAIRED_FUNGIBLE_ASSET));
+        assert!(!exists<PairedCoinType>(metadata_addr), error::already_exists(EPAIRED_COIN));
+        assert!(
+            coin_address<CoinType>() == signer::address_of(coin_creator),
```

Comment:
This potentially wouldn't work for most coins as the specific coin struct might have been deployed to a resource account with no upgradability. What are the considerations here for requiring this vs having MintCap & BurnCap?

Reply:
This function was supposed to be called by someone that only has FA initiated and then realize they need a coin. 
Having MintCap/BurnCap does not necessarily means the creator status.
for an existing coin, the FA can be setup by the framework automatically. They should not use this function.

Reply:
I don't think we need this... we should consider that either:
1. They can auto-upgrade and they can swap the capabilities for refs
2. They cannot upgrade and they go through the auto upgrade and suffer